src/Architectures.jl

```diff
@@ -39,12 +39,21 @@ Run Oceananigans on a single NVIDIA CUDA GPU.
 """
 struct GPU <: AbstractSerialArchitecture end
 
+"""
+    ReactantState <: AbstractArchitecture
+
+Run Oceananigans on Reactant.
+"""
+struct ReactantState <: AbstractSerialArchitecture end
+
 #####
 ##### These methods are extended in DistributedComputations.jl
 #####
 
 device(::CPU) = KernelAbstractions.CPU()
 device(::GPU) = CUDA.CUDABackend(; always_inline=true)
+# While there is no Reactant backend this worls
+device(::ReactantState) = CUDA.CUDABackend(; always_inline=true)
```

Comment:
Does it make sense to let the backend be a property of ReactantState?

Reply:
Currently this is used to say we are importing CUDA kernel source code to reactant, which is the current correct default (and should be decided in the KA reactant backend when ready).

Reply:
ok I see, I had a misconception about the meaning of "backend" and "device"

Reply:
I guess a comment would be appropriate here to describe what is happening

Reply:
Added comment